### PR TITLE
Add ‘Issue Type’ condition for report

### PR DESCRIPTION
**Pull request by roberttravispierce**
Resolves: https://jira.devops.va.gov/browse/APPEALS-30939
Main Jira effort: https://jira.devops.va.gov/browse/APPEALS-26499

# Description
Add the `Issue Type` condition field input for the Generate Task Report effort.

## Acceptance Criteria
- [ ] If user selects ‘Issue Type’ condition, issue types selection drop-down appears.
- [ ] Issue Type data will be populated using new JSON/constants file created in [APPEALS-30949](https://jira.devops.va.gov/browse/APPEALS-30949)
- [ ] User can type to find Issue Type name or select from drop-down.
- [ ] Drop-down options for ‘Issue Type’ include: list of VHA approved facilities (see below))
- [ ] User can type to search or select more options from the drop-down selection to include all desired options.
- [ ] User can remove any selected Issue Types displayed via the 'X' next to each selection. 
- [ ] At least one Issue Type must be selected if condition is added

## Testing Plan
- Log in as VHAADMIN1
- Go to decision_reviews/vha/report
- Click Add Condition
- Select Issue Type
- See the dropdown and interact with it
- Click submit and check the dev tools console to see if the correct data was logged

### Screenshots
Screenshot 2023-12-01 at 3 59 56 PM

### CODE REVIEW

File: client/app/nonComp/components/ReportPage/Conditions/IssueType.jsx [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
+  return {
+    value: issueType[0],
+    label: issueType[1]
+  };
+}).
+  sort((stringA, stringB) => stringA.label.localeCompare(stringB.label));
+
+export const IssueType = ({ control, field, name }) => {
+  const { errors } = useFormContext();
+  const nameIssueTypeCodes = `${name}.options.issueTypeCodes`;
```
Comment (TylerBroyles) [inline client/app/nonComp/components/ReportPage/Conditions/IssueType.jsx:24]: This name probably needs to match up everywhere with the error codes and schema so it should probably be changed to issueTypes or the errors schema needs to be changed to issueTypeCodes: errorStuff
```suggestion
  const nameIssueTypeCodes = `${name}.options.issueTypes`;
```

File: client/app/nonComp/components/ReportPage/Conditions/IssueType.jsx [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+
+  return (
+    <div className="issue-type-container">
+      <Controller
+        control={control}
+        name={nameIssueTypeCodes}
+        defaultValue={field.options.issueTypeCodes ?? []}
```
Comment (TylerBroyles) [inline client/app/nonComp/components/ReportPage/Conditions/IssueType.jsx:31]: Same thing as above
```suggestion
        defaultValue={field.options.issueTypes ?? []}
```

**Review by TylerBroyles — CHANGES_REQUESTED** [review]
I left some comments. Let me know what you think.

File: client/app/styles/_noncomp.scss [hunk 3]
```diff
@@ -183,48 +183,8 @@ $validation-error-color: #cd2026;
...
-        flex-direction: column;
-        padding: 16px;
-      }
-
-      .decisionReviewTypeContainerError {
-        border-left: solid $validation-error-color 4px;
-      }
-
-      .report-page-multi-select-dropdown {
+      .issue-type-container {
```
Comment (brandondorner) [inline client/app/styles/_noncomp.scss:186]: If we use `report-page-multi-select-dropdown` we don't need `issue-type-container`

File: client/test/app/nonComp/pages/ReportPage.test.js [hunk 4]
```diff
@@ -258,17 +253,56 @@ describe('ReportPage', () => {
...
+      expect(results).toHaveNoViolations();
+    });
+
+    it('renders an error if no selection is made', async () => {
+      await clickOnReportType();
+      await navigateToConditionInput('Issue Type');
+
+      const generateTaskReport = screen.getByRole('button', { name: 'Generate task report' });
+
+      // console.log(generateTaskReport);
```
Comment (brandondorner) [inline client/test/app/nonComp/pages/ReportPage.test.js:288]: There's a few stray comments in here that could be removed

**Review by brandondorner — CHANGES_REQUESTED** [review]

**Review by brandondorner — APPROVED** [review]

**Review by brandondorner — APPROVED** [review]

**Review by craigrva — APPROVED** [review]

**Pull request merged**